### PR TITLE
Fix highlighter issues

### PR DESCRIPTION
**Pull request by nightwing**
@lennartcl

### CODE REVIEW

File: lib/ace/edit_session/bracket_match.js [hunk 1]
```diff
@@ -117,7 +117,7 @@ function BracketMatch() {
             typeRe = new RegExp(
                 "(\\.?" +
                 token.type.replace(".", "\\.").replace("rparen", ".paren")
-                    .replace(/\b(?:end|start|begin)\b/, "")
```
Comment (lennartcl) [inline lib/ace/edit_session/bracket_match.js]: What does that do exactly? Without the `\b`'s it's much more general?

File: tool/lib.js [hunk 2]
```diff
@@ -11,9 +12,11 @@ exports.parsePlist = function(xmlOrJSON, callback) {
         plist.parseString(xmlOrJSON, function(_, result) {
             json = result[0];
         });
-    } else {
+    } else try {
```
Comment (lennartcl) [inline tool/lib.js]: That indentation looks a bit too irregular to me. It kinda works but to the layman might even suggest something like the catch applies to the whole if/else or something. It just looks a bit odd. I would just do brackets in the `else`.

**Comment by lennartcl** [comment]
Minor questions/comments, otherwise LGTM.

File: lib/ace/edit_session/bracket_match.js (hunk 1 above)
Comment (nightwing) [inline lib/ace/edit_session/bracket_match.js]: Modes imported from textmate have token types like `punctuation.begin.languagename`  `punctuation.start.langaugename` `punctuation.end.langaugename`. 
This is meant to create a regexp `punctuation\.(?:start|begin)\.languagename` but i see there's still a  bug, correct version would be `punctuation\.(?:start|begin|end)\.languagename`

**Pull request merged**